app/assets/javascripts/provider/admin/apiconfig/services/integration.js.coffee

```diff
@@ -208,17 +208,21 @@ $(document).on 'initialize', '#proxy', ->
     false
 
   proxy_credentials_location_value = $('input[name="proxy[credentials_location]"]:checked').val()
+  credentials_forms = document.querySelectorAll('li[id^="proxy_auth_"]')
+  credentials_forms.forEach((el) -> el.classList.toggle('hidden', proxy_credentials_location_value is 'authorization'))
```

Comment:
Besides refactoring to a single function, you may also check if `credentials_forms.length > 0`

Reply:
Well IMO the forms ids exist by definition, meaning you can take them for granted. Therefore, we don't need to check if they exist because even if they don't, that means the code is broken and there will be an error anyway. Or maybe I'm thinking too much here?